proto/desmos/reactions/v1/query.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+package desmos.reactions.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "cosmos_proto/cosmos.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+import "desmos/reactions/v1/models.proto";
+
+option go_package = "github.com/desmos-labs/desmos/v3/x/reactions/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Reactions allows to query the reactions present inside a subspace
+  rpc Reactions(QueryReactionsRequest) returns (QueryReactionsResponse) {
+    option (google.api.http).get =
+        "/desmos/reactions/v1/{subspace_id}/reactions";
+  }
+
+  // RegisteredReactions allows to query the registered reaction of a subspace
+  rpc RegisteredReactions(QueryRegisteredReactionsRequest)
+      returns (QueryRegisteredReactionsResponse) {
+    option (google.api.http).get =
+        "/desmos/reactions/v1/{subspace_id}/registered-reactions";
+  }
+
+  // ReactionsParams allows to query the reaction params of a subspace
+  rpc ReactionsParams(QueryReactionsParamsRequest)
+      returns (QueryReactionsParamsResponse) {
+    option (google.api.http).get = "/desmos/reactions/v1/{subspace_id}/params";
+  }
+}
+
+// QueryReactionsRequest is the request type for the Query/Reactions RPC method
+message QueryReactionsRequest {
+  // Id of the subspace to query the reactions for
+  uint64 subspace_id = 1;
+
+  // (optional) Post id to query the reactions for
+  uint64 post_id = 2;
+
```

Comment:
It would be useful to add `user` field that allows to filter by user.

Reply:
This is extremely hard to implement as currently reactions are stored using the following key: 
```
ReactionPrefix | SubspaceID | PostID | ReactionID -> ProtocolBuffer(Reaction) 
```

This is done in order to allow a faster removal of one reaction from a post (which is going to be the most likely operation to be carried out). 

If we want to allow querying a reaction for a user, we would have to somehow store the user address inside the store key, to be able to iterate over them. This is going to later complicate the retrieval of a specific reaction. 

For this reasons, I believe we can simply let third party tools (e.g. DJuno) to deal with this.

Reply:
Instead of storing the user address, It can be easily implemented by `FilteredPaginate` in grpc_query, say:
```go
func (k Keeper) Reactions(ctx context.Context, request *types.QueryReactionsRequest) (*types.QueryReactionsResponse, error) {
        ...
	var reactions []types.Reaction
	pageRes, err := query.FilteredPaginate(reactionsStore, request.Pagination, func(key []byte, value []byte, accumulate bool) (bool, error) {
		var reaction types.Reaction
		if err := k.cdc.Unmarshal(value, &reaction); err != nil {
			return false, status.Error(codes.Internal, err.Error())
		}

		if request.SubspaceId != 0 && request.PostId != 0 && request.User != "" && request.User != reaction.Author {
			return false, nil
		}

		reactions = append(reactions, reaction)
                if accumulate {
			reactions = append(reactions, reaction)
		}
		return true, nil
	})
       ...
}
```
Although it has `O(N)` time complexity, we can limit the `user` field to be only available when `SubspaceID` and `PostID` are given to make sure the amount `N` no to be large.

Reply:
I didn't think of using `FilteredPaginate`, you're right! I've implemented it this way now by also requiring users to specify a non-zero post id (it was useless to get all reactions inside a subspace since they're strongly tied to a post anyway)